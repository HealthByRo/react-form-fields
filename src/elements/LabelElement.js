/* @flow */

import React from 'react';
import type { LabelElementProps } from './types';

const LabelElement = (props: LabelElementProps) => (
  <label htmlFor={props.inputId}>
    {props.children}
    {props.label}
  </label>
);

export default LabelElement;
