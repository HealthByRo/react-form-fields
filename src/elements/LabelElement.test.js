/* @flow */
/*

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import LabelElement from './LabelElement';
import type { LabelElementProps } from './types';

describe('<LabelElement />', () => {
  let wrapper;
  const defaultProps = {
    htmlFor: 'INPUT_ID',
    text: 'FIELD_LABEL',
  };
  const shallowWithProps = (props: LabelElementProps = defaultProps) => {
    wrapper = shallow(
      <LabelElement {...props} />
    );
  };
  const AnyChild = () => <div>AnyChild</div>;

  describe('html element', () => {
    it('should be label', () => {
      shallowWithProps();

      expect(wrapper.name()).toBe('label');
    });

    it('should have htmlFor attribute', () => {
      shallowWithProps();

      expect(wrapper.props().htmlFor).toBeDefined();
      expect(wrapper.props().htmlFor).toEqual('INPUT_ID');
    });
  });

  describe('content', () => {
    it('should have FIELD_LABEL text', () => {
      shallowWithProps();

      expect(wrapper.text()).toEqual('FIELD_LABEL');
    });

    it('should have {children} if passed', () => {
      wrapper = shallow(
        <LabelElement {...defaultProps}>
          <AnyChild />
        </LabelElement>
      );

      expect(wrapper.find(AnyChild).length).toBe(1);
    });

    it('should render {children} before label text', () => {
      wrapper = renderer.create(
        <LabelElement {...defaultProps}>
          <AnyChild />
        </LabelElement>
      );

      expect(wrapper.toJSON()).toMatchSnapshot();
    });
  });
});
*/
