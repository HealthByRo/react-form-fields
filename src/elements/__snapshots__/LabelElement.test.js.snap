// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LabelElement /> content should render {children} before label text 1`] = `
<label
  htmlFor="INPUT_ID"
>
  <div>
    AnyChild
  </div>
  FIELD_LABEL
</label>
`;
